addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/api/model/UniFiController.java

```diff
@@ -0,0 +1,300 @@
+            }
+        }
+        return site;
+    }
+
+    // Device API
+
+    public @Nullable UniFiDevice getDevice(@Nullable String id) {
+        UniFiDevice device = null;
+        if (StringUtils.isNotBlank(id)) {
+            synchronized (this) {
+                device = devicesCache.get(id);
+            }
+            if (device == null) {
+                logger.debug("Could not find a matching device for id = '{}'", id);
+            }
+        }
+        return device;
+    }
+
+    // Client API
+
+    public @Nullable UniFiClient getClient(@Nullable String id) {
+        UniFiClient client = null;
+        if (StringUtils.isNotBlank(id)) {
+            synchronized (this) {
+                // mgb: first check active clients and fallback to insights if not found
+                client = clientsCache.get(id);
+                if (client == null) {
+                    client = insightsCache.get(id);
+                }
+            }
+            if (client == null) {
+                logger.debug("Could not find a matching client for id = {}", id);
+            }
+        }
+        return client;
+    }
+
+    protected void block(UniFiClient client, boolean blocked) throws UniFiException {
+        UniFiControllerRequest<Void> req = newRequest(Void.class);
+        req.setPath("/api/s/" + client.getSite().getName() + "/cmd/stamgr");
+        req.setBodyParameter("cmd", blocked ? "block-sta" : "unblock-sta");
+        req.setBodyParameter("mac", client.getMac());
+        executeRequest(req);
+    }
+
+    protected void reconnect(UniFiClient client) throws UniFiException {
+        UniFiControllerRequest<Void> req = newRequest(Void.class);
+        req.setPath("/api/s/" + client.getSite().getName() + "/cmd/stamgr");
+        req.setBodyParameter("cmd", "kick-sta");
+        req.setBodyParameter("mac", client.getMac());
+        executeRequest(req);
+    }
+
+    // Internal API
+
+    private <T> UniFiControllerRequest<T> newRequest(Class<T> responseType) {
+        return new UniFiControllerRequest<T>(responseType, gson, httpClient, host, port);
+    }
+
+    private <T> T executeRequest(UniFiControllerRequest<T> request) throws UniFiException {
+        T result;
+        try {
+            result = request.execute();
+        } catch (UniFiExpiredSessionException e) {
+            login();
+            result = executeRequest(request);
+        } catch (UniFiNotAuthorizedException e) {
+            logger.warn("Not Authorized! Please make sure your controller credentials have administrator rights");
+            result = null;
+        }
+        return result;
+    }
+
+    private UniFiSiteCache getSites() throws UniFiException {
+        UniFiControllerRequest<UniFiSite[]> req = newRequest(UniFiSite[].class);
+        req.setPath("/api/self/sites");
+        UniFiSite[] sites = executeRequest(req);
+        logger.debug("Found {} UniFi Site(s): {}", sites.length, lazyFormatAsList(sites));
```

Comment:
It looks like `executeRequest` can return null, which would lead to a null pointer exception here. Same on other places.